EventEmitter memory leak detected

As reported here https://github.com/feathersjs/client/issues/236

In my Angular app I've started to get the following error in certain circumstances and not sure why.

> feathers.js:2685 (node) warning: possible EventEmitter memory leak detected. 11 listeners added. Use emitter.setMaxListeners() to increase limit. 

My page has next and prev links that allow me to navigate to the next gallery but after the 11th navigation it gives the error. In my component below I'm just subscribing to the route change to then fetch the next or previous item

```
export class GalleryDetailComponent implements OnInit, OnDestroy {
  gallery$: Observable<any>;
  constructor(private service: ApiService, private route: ActivatedRoute) {
    service.setup('gallery');
  }

  ngOnInit() {
    this.getGallery();
  }

  getGallery() {
    this.gallery$ = this.route.params.pipe(
      switchMap((params: any) => {
        return this.service.get(params['slug']);
      })
    );
  }

  ngOnDestroy() {}
}
```

and this is my feathers service and api service.
```
@Injectable()
export class FeathersService {
  private _feathers: any;
  private _rest: any;
  constructor(public http: HttpClient) {
    this._rest = rest(environment.host);
    this._feathers = feathers();
    this._feathers.configure(
      authentication({
        storageKey: 'auth-token',
        storage: new CookieStorage({
          path: '/',
          expires: new Date(Date.now() + 30 * 24 * 60 * 60 * 1000)
        })
      })
    );
    this._feathers.configure(
      this._rest.angularHttpClient(this.http, { HttpHeaders: HttpHeaders })
    );
    this._feathers.configure(
      feathersRx({
        listStrategy: 'always',
        idField: '_id'
      })
    );
  }

  // expose services
  public service(name: string) {
    return this._feathers.service(name);
  }
}

```

```
@Injectable()
export class ApiService {
  private service: any;
  constructor(private feathers: FeathersService) {}

  setup(endpoint: string) {
    this.service = this.feathers.service(endpoint);
  }

  get(id: number | string, query?: any) {
    return this.service.watch({ strategy: 'always' }).get(id, { query: query });
  }
}
```

Unfortunately that version throws up a load of errors

```
Module not found: Error: Can't resolve 'feathers-reactive' in '/src/client/app/shared/services'
resolve 'feathers-reactive' in '/src/client/app/shared/services'
  Parsed request is a module
  using description file: /package.json (relative path: ./src/client/app/shared/services)
```

--- a/src/index.js
+++ b/src/index.js
@@ -1,6 +1,7 @@
 import _debug from 'debug';
 
 import { fromEvent } from 'rxjs';
+import { share } from 'rxjs/operators';
 import reactiveResource from './resource';
 import reactiveList from './list';
 import strategies from './strategies';
@@ -31,10 +32,10 @@ function FeathersRx (options = {}) {
       // fromEvent's result selector (3rd arg) is deprecated
       // we need it here because service events have an inconsistent number of arguments (i.e. sometimes 1, sometimes >1)
       // related: https://github.com/ReactiveX/rxjs/issues/3751
-      created: fromEvent(service, 'created', (...args) => args[0]),
-      updated: fromEvent(service, 'updated', (...args) => args[0]),
-      patched: fromEvent(service, 'patched', (...args) => args[0]),
-      removed: fromEvent(service, 'removed', (...args) => args[0])
+      created: fromEvent(service, 'created', (...args) => args[0]).pipe(share()),
+      updated: fromEvent(service, 'updated', (...args) => args[0]).pipe(share()),
+      patched: fromEvent(service, 'patched', (...args) => args[0]).pipe(share()),
+      removed: fromEvent(service, 'removed', (...args) => args[0]).pipe(share())
     };
 
     // object to hold our reactive methods

